src/db/DbService.js

```
@@ -131,36 +123,32 @@ class DbService {
+        const dataEntries = Object.entries(data);
 
-        Object.entries(data).forEach(([key, value]) => {
-            query += ` AND (dataKey0 = ? AND dataValue0 = ?) OR (dataKey1 = ? AND dataValue1 = ?) OR (dataKey2 = ? AND dataValue2 = ?) OR (dataKey3 = ? AND dataValue3 = ?) `;
+        dataEntries.forEach(([key, value]) => {
+            query += `(dataKey0 = ? AND dataValue0 = ?) OR (dataKey1 = ? AND dataValue1 = ?) OR (dataKey2 = ? AND dataValue2 = ?) OR (dataKey3 = ? AND dataValue3 = ?) `;
             queryParams.push(key, value, key, value, key, value, key, value);
         });
 
         if (address) {
-            query += ' AND user_wallet_address = ?';
+            query += (dataEntries.length ? 'AND' : '') + ' user_wallet_address = ?';
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)